src/handwritten/_lib_time.py

```diff
@@ -29,21 +30,29 @@ class AbsoluteTime(ctypes.Structure):
     MAX_FS = 10**9
     MAX_YS = 10**9
 
+    def convert_to_desired_timezone(self, expected_time_utc, tzinfo):
+        current_time_utc = ht_datetime.now(timezone.utc)
+        desired_timezone_offset = current_time_utc.astimezone(tz=tzinfo).utcoffset()
+        desired_expected_time = expected_time_utc + desired_timezone_offset
+        new_datetime = ht_datetime(
+            desired_expected_time.year,
+            desired_expected_time.month,
+            desired_expected_time.day,
+            desired_expected_time.hour,
+            desired_expected_time.minute,
+            desired_expected_time.second,
+            desired_expected_time.microsecond,
+            desired_expected_time.femtosecond,
+            desired_expected_time.yoctosecond,
+            tzinfo = tzinfo)
+        return new_datetime
+
     @classmethod
     def from_datetime(cls, dt: Union[std_datetime, ht_datetime]) -> AbsoluteTime:
-        utc_dt = dt.astimezone(tz=timezone.utc)
-
-        # First, calculate whole seconds by converting from the 1970 to 1904 epoch.
-        timestamp_1970_epoch = utc_dt.timestamp()
-        was_negative = timestamp_1970_epoch < 0
-        timestamp_1904_epoch = int(timestamp_1970_epoch + AbsoluteTime._BIAS_FROM_1970_EPOCH)
+        epoch_1904 = ht_datetime(1904, 1, 1, tzinfo=timezone.utc)
+        seconds_since_1904 = int(abs(dt - epoch_1904) / ht_timedelta(seconds=1))
```

Comment:
Why does the lib version divide by 1 second and the grpc version use `.total_seconds()`?

Reply:
I still don't understand why this is different.

Reply:
You are right, total_seconds() is the better option. I have updated it to use total_seconds()